mne/time_frequency/csd.py

```diff
@@ -1,63 +1,522 @@
+        object to compute a DICS beamformer. Defaults to None, which means no
+        projectors will be applied.
+    is_sum : bool
+        Whether the CSD matrix represents a sum (or average) across
+        frequencies. Defaults to False.
+
+    See Also
+    --------
+    csd_epochs
+    csd_array
     """
 
-    def __init__(self, data, ch_names, projs, bads, freqs,
-                 n_fft):  # noqa: D102
-        self.data = data
-        self.dim = len(data)
-        self.ch_names = cp.deepcopy(ch_names)
-        self.projs = cp.deepcopy(projs)
-        self.bads = cp.deepcopy(bads)
-        self.freqs = np.atleast_1d(np.copy(freqs))
+    def __init__(self, data, names, tmin, tmax, frequencies, n_fft, projs=None,
+                 is_sum=False):
+        data = np.asarray(data)
+        if data.ndim == 1:
+            data = data[:, np.newaxis]
+        elif data.ndim > 2:
+            raise ValueError('`data` should be either a 1D or 2D array.')
+        self._data = data
+
+        if len(names) != _n_dims_from_triu(len(data)):
+            raise ValueError('Number of names does not match the number of '
+                             'time series in the CSD matrix.')
+        self.names = names
+        self.tmin = tmin
+        self.tmax = tmax
+
+        if isinstance(frequencies, numbers.Number):
+            frequencies = [frequencies]
+        if len(frequencies) != data.shape[1]:
+            raise ValueError('Number of frequencies does not match the number '
+                             'of CSD matrices in the data array (%d != %d).' %
+                             (len(frequencies), data.shape[1]))
+        self.frequencies = frequencies
+
         self.n_fft = n_fft
+        self.projs = cp.deepcopy(projs)
+        self.is_sum = is_sum
+
+    @property
+    def n_series(self):
+        """Number of time series defined in this CSD object."""
+        return len(self.names)
 
     def __repr__(self):  # noqa: D105
-        s = 'frequencies : %s' % self.freqs
-        s += ', size : %s x %s' % self.data.shape
-        s += ', data : %s' % self.data
-        return '<CrossSpectralDensity  |  %s>' % s
+        # Make a pretty string representation of the frequencies
+        freq_strs = []
+        for f in self.frequencies:
+            if isinstance(f, numbers.Number):
+                freq_strs.append(str(f))
+            elif len(f) == 1:
+                freq_strs.append(str(f[0]))
+            else:
+                freq_strs.append('{}-{}'.format(np.min(f), np.max(f)))
+        freq_str = ', '.join(freq_strs) + ' Hz.'
+
+        return (
+            '<CrossSpectralDensity  |  '
+            'n_series={}, time={} to {} s, frequencies={}>'
+        ).format(self.n_series, self.tmin, self.tmax, freq_str)
+
+    def sum(self, fmin=None, fmax=None):
+        """Calculate the sum CSD in the given frequency range(s).
+
+        If the exact given frequencies are not available, the nearest
+        frequencies will be chosen.
```

Comment:
isn't it more natural to do mean?

Reply:
in other words why do you need sum and mean is not enough?

Reply:
Backwards compatibility. The original `csd_epochs` has a `fsum=True` parameter that sums across the frequencies. I agree that mean is more natural tough.